File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n=nums.size();
        if(n==1){
            return 1;
        }
        int l=0;
        int r=1;
        while(r<n){
            if(nums[l]==nums[r]){
                r++;
            }else{
                l++;
                nums[l]=nums[r];
            }
        }
        return l+1;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {

        LinkedHashMap<Integer, Integer> lHM = new LinkedHashMap<>(); 
        for(int i = 0; i < nums.length; i++)
        {
            lHM.put(nums[i], lHM.getOrDefault(nums[i], 0) + 1);


        }
        int i = 0;
        for(int elem : lHM.keySet())
        {
            nums[i++] = elem;

        }
        return lHM.size();
        
    }
}